magicctapipe/scripts/lst1_magic/lst1_magic_cloud_correction.py

```diff
@@ -0,0 +1,715 @@
+            )  # For LST-1 we compute concentration from the cleaned image and for MAGIC from the full image to reproduce the current behaviour in the standard code
+        else:
+            conc_params = concentration_parameters(camgeom, clean_image, hillas_params)
+
+        event_params = {
+            **hillas_params,
+            **timing_params,
+            **leakage_params,
+        }
+
+        prefixed_conc_params = {
+            f"concentration_{key}": value for key, value in conc_params.items()
+        }
+        event_params.update(prefixed_conc_params)
+
+        event_info_dict = {
+            "obs_id": obs_id,
+            "event_id": event_id,
+            "tel_id": tel_id,
+            "pointing_alt": pointing_alt,
+            "pointing_az": pointing_az,
+            "time_diff": time_diff,
+            "n_pixels": signal_pixels,
+            "n_islands": n_islands,
+            "event_id_lst": event_id_lst,
+            "obs_id_lst": obs_id_lst,
+            "event_id_magic": event_id_magic,
+            "obs_id_magic": obs_id_magic,
+            "combo_type": combo_type,
+            "timestamp": timestamp,
+            "multiplicity": multiplicity,
+        }
+        event_params.update(event_info_dict)
+
+        all_params_list.append(event_params)
+
+    df = pd.DataFrame(all_params_list)
+    return df
+
+
+def main():
+    """Main function."""
+    start_time = time.time()
+    parser = argparse.ArgumentParser()
+
+    parser.add_argument(
+        "--input_file",
+        "-i",
+        dest="input_file",
+        type=str,
+        required=True,
+        help="Path to an input .h5 DL1 data file",
+    )
+
+    parser.add_argument(
+        "--output_dir",
+        "-o",
+        dest="output_dir",
+        type=str,
+        default="./data",
+        help="Path to a directory where to save an output corrected DL1 file",
+    )
+
+    parser.add_argument(
+        "--config_file",
+        "-c",
+        dest="config_file",
+        type=str,
+        default="./resources/config.yaml",
+        help="Path to a configuration file",
+    )
+
+    args = parser.parse_args()
+
+    subarray_info = SubarrayDescription.from_hdf(args.input_file)
+
+    tel_descriptions = subarray_info.tel
+    camgeom = {}
+    for telid, telescope in tel_descriptions.items():
+        camgeom[telid] = telescope.camera.geometry
+
+    optics_table = read_table(
+        args.input_file, "/configuration/instrument/telescope/optics"
+    )
+    focal_eff = {}
+
+    for telid, telescope in tel_descriptions.items():
+        optics_row = optics_table[optics_table["optics_name"] == telescope.name]
+        if len(optics_row) > 0:
+            focal_eff[telid] = optics_row["effective_focal_length"][0] * u.m
+        else:
+            raise ValueError(f"No optics data found for telescope: {telescope.name}")
+
+    with open(args.config_file, "r") as file:
+        config = yaml.safe_load(file)
+
+    tel_ids = config["mc_tel_ids"]
```

Comment:
why do you get them from here? those are MC tel_ids, you should be able to use the tel_descriptions above to figure out what telescope you are looking at now